src/client/list.rs

```diff
@@ -42,86 +42,44 @@ pub(crate) trait ListClientExt {
     fn list_paginated(
         &self,
         prefix: Option<&Path>,
-        delimiter: bool,
-        offset: Option<&Path>,
+        opts: ListOpts,
     ) -> BoxStream<'static, Result<ListResult>>;
-
-    fn list(&self, prefix: Option<&Path>) -> BoxStream<'static, Result<ObjectMeta>>;
-
-    #[allow(unused)]
-    fn list_with_offset(
-        &self,
-        prefix: Option<&Path>,
-        offset: &Path,
-    ) -> BoxStream<'static, Result<ObjectMeta>>;
-
-    async fn list_with_delimiter(&self, prefix: Option<&Path>) -> Result<ListResult>;
 }
 
 #[async_trait]
 impl<T: ListClient + Clone> ListClientExt for T {
```

Comment:
Can we add a list_opts impl here, to avoid duplicating the plumbing logic in the 3 cloud provider stores

Reply:
do you mean add a `list_opts` and then calling `list_paginated` within `list_opts` impl, and the 3 clould provider stores call `self.client.list_opts(prefix, options)`? like this:
```
fn list_opts(
    &self,
    prefix: Option<&Path>,
    opts: ListOpts,
) -> BoxStream<'static, Result<ListResult>> {
    let client = self.clone();
    client.list_paginated(prefix, opts)
}
```
The 3 clould provider are calling `list_paginated` method directly during implementing `list_opts` of ObjectStore, because the `list_opts` of `ListClientExt` will be very simple, just transfer request.

Reply:
I'm not entirely sure I follow what you are saying, but if list_paginated ends up with the same signature as list_opts, it might as well just get renamed to list_opts.